distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso

```diff
@@ -89,16 +89,30 @@ prepare_aggregate_columns aggregates table =
 
    Assumes that the `Aggregate_Column` is resolved. You may need to transform it
    with `resolve_aggregate` first.
-default_aggregate_column_name aggregate_column =
+default_aggregate_column_name aggregate_column include_column=True =
     case aggregate_column of
         Group_By c _ -> c.name
         Count _ -> "Count"
         Count_Distinct columns _ _ ->
             "Count Distinct " + (columns.map .name . join " ")
-        Percentile p c _ -> ((p*100).floor.to_text + "%-ile ") + c.name
+        Percentile p c _ -> ((p*100).floor.to_text + "%-ile ") + (if include_column then c.name else "")
         _ ->
             prefix = Meta.get_simple_type_name aggregate_column . replace "_" " "
-            prefix + " " + aggregate_column.column.name
+            c = aggregate_column.column
+            prefix + " " + (if include_column then c.name else "")
+
+## Utility function to check if all same column
+all_same_column : Vector Aggregate_Column -> Boolean
+all_same_column aggregates =
+    is_not_count c = case c of
+        Count _ -> False
+        Count_Distinct _ _ _ -> False
+        _ -> True
+    without_count = aggregates.filter is_not_count
+
+    if without_count.length < 2 then True else
+        column = without_count.first.column.name
+        without_count.all c->(c.column.name == column)
```

Comment:
It's usually easier to follow if we have a check that simply call `.is_empty`. Plus `.length` call can seem a bit spurious from the performance POV.

Reply:
Isn't the performance of `length` and `is_empty` basically the same? Under the hood `is_empty = length > 0` IIRC for Vector anyway.

Reply:
Oh. That's a bummer.

Reply:
In general _every_ `.length` on any collection should be a very fast operation.
But it is spurious when you only want to check if it is empty or not.

Reply:
Hm I guess it may not necessarily be fast on a `List` (whereas `is_empty` will still be fast there). But my point is that in case of Vector it just boils down to reading its property. And that there's just no other way to check if it's empty than to check that `length` property in that case. But of course for other collections (like `List`) there can be a difference.

Reply:
> But my point is that in case of Vector it just boils down to reading its property

Sure, no disagreement on that. My comment was more generic since I didn't even check if `without_count` is a `Vector` or `List` or something else. I just knew that it was some collection.
The discussion about `is_empty` implementation is a slightly orthogonal subject not relevant for this PR.

Going back to my comment, my preference is just that when I see `length` being compared to some number I have to make a mental note why this number is used. If I see `is_empty` it is immediately obvious what the author had in mind. This is just a minor nit.

Reply:
> The discussion about `is_empty` implementation is a slightly orthogonal subject not relevant for this PR.

Yep, definitely :)

> Going back to my comment, my preference is just that when I see length being compared to some number I have to make a mental note why this number is used. If I see is_empty it is immediately obvious what the author had in mind. This is just a minor nit.

I agree 100% here